### PR TITLE
Amit/roger 1592

### PR DESCRIPTION
**Pull request by amitbose327**

### CODE REVIEW

File: ansible/roles/generate-permission-files/files/generate_nginx_basic_auth_file.py [hunk 1]
```diff
@@ -7,12 +7,16 @@
 import yaml
 import os
 
-def generate_http_basic_auth_file(permissions_file, output_file):
-    if not permissions_file:
+def generate_http_basic_auth_file(permissions_files, output_file):
```
Comment (markg-moz) [inline ansible/roles/generate-permission-files/files/generate_nginx_basic_auth_file.py]: The filename and function name are different.

**Comment by markg-moz** [comment]
LGTM

File: ansible/roles/generate-permission-files/files/generate_nginx_basic_auth_file.py [hunk 2]
```diff
@@ -23,10 +27,24 @@ def generate_http_basic_auth_file(permissions_file, output_file):
...
+    else:
+        dict1 = dict2 # latter value overrides
+    return dict1
             
 if __name__ == '__main__':
     if len(sys.argv) < 3:
-        sys.exit("Usage: %s <permissions_file> <output_file>" % sys.argv[0])
-    permissions_file = sys.argv[1]
+        sys.exit("Usage: %s <permissions_files_delimited_by_space> <output_file>" % sys.argv[0])
+    permissions_files = sys.argv[1]
```
Comment (vmahedia) [inline ansible/roles/generate-permission-files/files/generate_nginx_basic_auth_file.py]: how will this work if it is delimited by space?

Comment (amitbose327) [inline ansible/roles/generate-permission-files/files/generate_nginx_basic_auth_file.py]: This was a typo. The files are delimited by commas.

File: ansible/roles/generate-permission-files/files/generate_nginx_basic_auth_file.py [hunk 3]
```diff
@@ -23,10 +27,24 @@ def generate_http_basic_auth_file(permissions_file, output_file):
             type = user_data.get('type', 'user')
             if type == "user":
                 htpasswd_file.add(username, username)
+
+def merge_dicts(dict1, dict2):
+    ''' Merge dict2 into dict1 recursively, merging lists for the same keys, overriding with the latter value (for non-list values), and return updated dict1 '''
```
Comment (amitbose327) [inline ansible/roles/generate-permission-files/files/generate_nginx_basic_auth_file.py]: Simplified the code.

**Pull request merged**